qemu/tests/blockdev_commit_general_operation.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from provider import job_utils
+from provider import backup_utils
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitGeneralOperation(BlockDevCommitTest):
+
+    def commit_op(self, cmd, args=None):
+        self.main_vm.monitor.cmd(cmd, args)
+        job_status = job_utils.query_block_jobs(self.main_vm)
+        logging.info("The status after operation %s is %s", cmd, job_status)
+
+    def commit_snapshots(self):
+        device = self.params.get("device_tag")
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        options = ["speed", "auto-finalize", "auto-dismiss"]
+        arguments = self.params.copy_from_keys(options)
+        arguments["speed"] = self.params["commit_speed"]
+        arguments["auto-finalize"] = self.params.get_boolean("finalize")
+        arguments["auto-dismiss"] = self.params.get_boolean("dismiss")
+        device = self.get_node_name(snapshot_tags[-1])
+        commit_cmd = backup_utils.block_commit_qmp_cmd
+        cmd, args = commit_cmd(device, **arguments)
+        self.main_vm.monitor.cmd(cmd, args)
+        job_id = args.get("job-id", device)
+        self.main_vm.monitor.cmd("block-job-set-speed",
+                                 {'device': job_id, 'speed': 10240})
+        self.commit_op("stop")
+        self.commit_op("cont")
+        self.commit_op("job-pause", {'id': job_id})
+        self.commit_op("job-resume", {'id': job_id})
+        self.commit_op("job-cancel", {'id': job_id})
```

Comment:
On ppc, following test might fail since the 'job-cancel' has not finished, so suggest to add 2 seconds sleep time.

03:35:43 INFO | The status after operation job-cancel is [{'auto-finalize': True, 'io-status': 'ok', 'device': 'drive_sn1_WPxL', 'auto-dismiss': True, 'busy': True, 'len': 11206656, 'offset': 524288, 'status': 'running', 'paused': False, 'speed': 10240, 'ready': False, 'type': 'commit'}]

Reply:
what about wait the specific job cancel event instead of waiting for some seconds

Reply:
Okay,will update it.